src/opsinputs/opsinputs_varobswriter_mod.F90

```diff
@@ -1213,54 +1210,104 @@ end subroutine opsinputs_varobswriter_fillreportflags
 !> e.g. for HIRS & AMSUA 
 !> the number of these channels is stored in Ob % NumChans.
 subroutine opsinputs_varobswriter_fillchannumandnumchans( &
-  Ob, ObsSpace, Channels, Flags, FillChanNum, FillNumChans, &
-  OffsetChans,UseActualChans)
+  Ob, ObsSpace, channels, varChannels, Flags, FillChanNum, FillNumChans, compressVarChannels, &
+  varObsSize, increaseChanArray)
+
 implicit none
 
 ! Subroutine arguments:
 type(OB_type), intent(inout)   :: Ob
 type(c_ptr), value, intent(in) :: ObsSpace
-integer(c_int), intent(in)     :: Channels(:)
+integer(c_int), intent(in)     :: channels(:)
+integer(c_int), intent(in)     :: varChannels(:)
 type(c_ptr), value, intent(in) :: Flags
 logical, intent(in)            :: FillChanNum, FillNumChans
-integer, intent(in)            :: OffsetChans
-logical, optional, intent(in)  :: UseActualChans
+logical                     :: compressVarChannels
+integer(integer64), optional, intent(in) :: varObsSize
+logical, optional, intent(in)  :: increaseChanArray
+
 
 ! Local declarations:
 integer(integer64)             :: NumChannels
-integer(integer64)             :: ChannelIndices(Ob % Header % NumObsLocal, size(Channels))
+integer(integer64), allocatable            :: ChannelIndicesVar(:,:)
+integer(integer64)             :: ChannelIndices(Ob % Header % NumObsLocal, size(channels))
 integer(integer64)             :: ChannelCounts(Ob % Header % NumObsLocal)
 integer                        :: iChannel
 integer                        :: iObs
-logical                        :: localUseActualChans
+real(kind=c_double)                             :: MissingDouble
+integer(integer64)             ::array_loop
+
+integer                        :: offsetsize
 
 ! Body:
-NumChannels = size(Channels)
+
+MissingDouble = missing_value(0.0_c_double)
+
+NumChannels = size(channels)
 if (NumChannels == 0) return
 
-! Setup for useActualChans for array indices
-localUseActualchans = .false.
-if (Present(UseActualChans)) then
-  localUseActualChans = UseActualChans
+if (increaseChanArray) then
```

Comment:
again this routine is not safe with its use of optional arguments without checking they are present.  I would suggest creating local variables from all optional.